app/policies/task_contribution_policy.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class TaskContributionPolicy < ApplicationPolicy
+  def task_contribution
+    @record
+  end
+
+  def suggestion
+    @record.suggestion
+  end
+
+  def base
+    @record.base
+  end
+
+  %i[create? new?].each do |action|
+    define_method(action) do
+      Pundit.policy(@user, base).contribute?
+    end
+  end
+
+  # While discard_changes? and show? look similar, the difference in paraphrases position is relevant
```

Comment:
What are you referring to with `paraphrases`?

```suggestion
  # While discard_changes? and show? look similar, the difference in the position of parentheses is relevant
```